File: app/src/main/java/com/lmyby/ankihelper/data/ai/AIException.java
package com.lmyby.ankihelper.data.ai;

import org.json.JSONObject;

/**
 * Custom exception class for AI-related errors
 */
public class AIException extends Exception {
    private AIErrorType errorType;
    private Throwable cause;
    
    public AIException(AIErrorType errorType, String message) {
        super(message);
        this.errorType = errorType;
    }
    
    public AIException(AIErrorType errorType, String message, Throwable cause) {
        super(message, cause);
        this.errorType = errorType;
        this.cause = cause;
    }
    
    public AIErrorType getErrorType() {
        return errorType;
    }
    
    @Override
    public Throwable getCause() {
        return cause;
    }
    
    /**
     * Converts the exception to a JSON error response
     * 
     * @return JSON string representation of the error
     */
    public String toJsonResponse() {
        try {
            JSONObject errorResponse = new JSONObject();
            JSONObject errorObj = new JSONObject();
            
            errorObj.put("type", errorType.name().toLowerCase());
            errorObj.put("message", getMessage());
            errorObj.put("details", cause != null ? cause.getMessage() : "");
            
            errorResponse.put("error", errorObj);
            
            return errorResponse.toString();
        } catch (Exception e) {
            return "{\"error\":{\"type\":\"unknown\",\"message\":\"Failed to serialize error\"}}";
        }
    }
}

File: app/src/main/java/com/lmyby/ankihelper/data/ai/AIException.kt
package com.lmyby.ankihelper.data.ai

import org.json.JSONObject

/**
 * Custom exception class for AI-related errors
 * Converted to Kotlin as part of Phase 2 data model migration
 */
class AIException : Exception {
    val errorType: AIErrorType
    override val cause: Throwable?

    constructor(errorType: AIErrorType, message: String) : super(message) {
        this.errorType = errorType
        this.cause = null
    }

    constructor(errorType: AIErrorType, message: String, cause: Throwable) : super(message, cause) {
        this.errorType = errorType
        this.cause = cause
    }

    /**
     * Converts the exception to a JSON error response
     *
     * @return JSON string representation of the error
     */
    fun toJsonResponse(): String {
        return try {
            val errorResponse = JSONObject()
            val errorObj = JSONObject()

            errorObj.put("type", errorType.name.lowercase())
            errorObj.put("message", message)
            errorObj.put("details", cause?.message ?: "")

            errorResponse.put("error", errorObj)

            errorResponse.toString()
        } catch (e: Exception) {
            "{\"error\":{\"type\":\"unknown\",\"message\":\"Failed to serialize error\"}}"
        }
    }
}
